ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -22,23 +23,35 @@
 #import "CKMacros.h"
 #import "CKThreadLocalComponentScope.h"
 
+static bool keyVectorsEqual(const std::vector<id<NSObject>> &a, const std::vector<id<NSObject>> &b)
+{
+  if (a.size() != b.size()) {
+    return false;
+  }
+  return std::equal(a.begin(), a.end(), b.begin(), [](id<NSObject> x, id<NSObject> y){
+    return CKObjectIsEqual(x, y) == YES; // be pedantic here becuase BOOL != bool
+  });
+}
```

Comment:
Decided to skip crazy template specialization stuff here. I want the explicitness of definitely using CKObjectIsEqual.